functional-tests/src/test/java/org/infinispan/online/service/endpoint/HotRodTester.java

```diff
@@ -34,20 +36,130 @@ public void testPutPerformance(URL urlToService, long timeout, TimeUnit timeUnit
       TestObjectCreator testObjectCreator = new TestObjectCreator();
 
       //when
-      while(System.currentTimeMillis() - endTime < 0) {
+      while (System.currentTimeMillis() - endTime < 0) {
          String key = testObjectCreator.getRandomString(1000);
          String value = testObjectCreator.getRandomString(1000);
          defaultCache.put(key, value);
       }
    }
 
-   private RemoteCache<String, String> getDefaultCache(URL urlToService) {
+   public RemoteCache<String, String> getDefaultCache(URL urlToService) {
+      RemoteCacheManager cachingService = getRemoteCacheManager(urlToService);
+      return cachingService.getCache();
+   }
+
+   public RemoteCache<String, String> getNamedCache(URL urlToService, String cacheName) {
+      RemoteCacheManager cachingService = getRemoteCacheManager(urlToService);
+      return cachingService.getCache(cacheName);
+   }
+
+   public RemoteCacheManager getRemoteCacheManager(URL urlToService) {
+      //given
       Configuration cachingServiceClientConfiguration = new ConfigurationBuilder()
          .addServer()
          .host(urlToService.getHost())
          .port(urlToService.getPort())
          .build();
-      RemoteCacheManager cachingService = new RemoteCacheManager(cachingServiceClientConfiguration);
-      return cachingService.getCache();
+
+      return new RemoteCacheManager(cachingServiceClientConfiguration);
+   }
+
+   public void putGetTest(URL hotRodService) {
```

Comment:
I really don't like this test! There's a couple of things wrong in it:

1. We already do those tests in Infinispan. Does it really make sense to test it here as well?
2. Writing "story" type of tests (do something, assert, do something else, assert etc) is horrible to debug if something goes wrong in the middle of it. Please test one scenario per test.
3. Please use //given//when//then convention as we do in other tests. See this nice write-up as a reference: https://monkeyisland.pl/2009/12/07/given-when-then-forever/

Reply:
Hey Sebastian,
so what about testing just String, byte[] and one primitive type? The first two will be important in "compatibility" tests between clients but I think it wouldn't hurt to have them here as well. Since the server side can be configured differently than it is in other test suites we can't ignore all data types and use just one.
These tests are mentioned in the test plan https://docs.google.com/document/d/1p5v5HerE6W2JucIIuDhUk5LuJCILjHOdIacSU97Lp8A/edit?ts=59f7384e so if you think some scenarios are unnecessary to test please comment there as well before we get to them.

Reply:
Right, compatibility can be tricky since it will be replaced by transcoding... The point is that I would really love to avoid put&get&assert spaghetti at all cost.

So in order to create a fair test, we need to test all 3 combinations of `put`s and `get`s right (E.g. put via 'Hot Rod' and get via 'REST'). I would add a generic `put` and `get` method to the endpoint testers (they already expose the same interface) and create a parametrized test with all the combinations. This would also be a nice occasion to refactor (or remove) `testBasicEndpointCapabilities` method and update other tests so that they look consistent with the new ones.

Reply:
OK. I generally agree. But I'm not sure the parametrized test will work when we move the test to real OpenShift. Running REST and HotRod inside OpenShift and outside OpenShift is different for both clients. REST can be made available via Router while HotRod will be available via NodePort or some other way (it will need different setup). So we might need to do some refactoring later.

Reply:
Yes, that's right...

Maybe we should think about moving our automatic test-suite from `oc cluster up` to `minishift`? The latter uses VM so we should have very similar experience to a "real" OpenShift installation?

//cc @ryanemerson